fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -282,6 +282,8 @@ private PlanNode createQueryPlan(QueryStmt stmt, Analyzer analyzer, long default
             }
         }
 
+        long sqlSelectLimit = analyzer.getContext().getSessionVariable().sqlSelectLimit;
```

Comment:
createQueryPlan is called recursively, should make sqlSelectLimit as a parameter of this function just like defaultOrderByLimit